saleor/tests/e2e/checkout/discounts/sales/test_checkout_product_on_percentage_sale.py

```diff
@@ -113,47 +166,225 @@ def test_checkout_products_on_percentage_sale_core_1004(
+    calculated_total = calculated_subtotal + shipping_price
     total_gross_amount = checkout_data["totalPrice"]["gross"]["amount"]
     subtotal_gross_amount = checkout_data["subtotalPrice"]["gross"]["amount"]
+    assert total_gross_amount == calculated_total
+    assert subtotal_gross_amount == calculated_subtotal
 
-    # Step 3 - Create payment for checkout.
+    # Step 6 - Create payment for checkout.
     checkout_dummy_payment_create(
         e2e_not_logged_api_client,
         checkout_id,
         total_gross_amount,
     )
 
-    # Step 5 - Complete checkout.
+    # Step 7 - Complete checkout.
     order_data = checkout_complete(
         e2e_not_logged_api_client,
         checkout_id,
     )
     assert order_data["status"] == "UNFULFILLED"
-    assert order_data["total"]["gross"]["amount"] == total_gross_amount
-    assert order_data["subtotal"]["gross"]["amount"] == subtotal_gross_amount
+    assert order_data["total"]["gross"]["amount"] == calculated_total
+    assert order_data["subtotal"]["gross"]["amount"] == calculated_subtotal
 
-    order_line = order_data["lines"][0]
-    assert order_line["undiscountedUnitPrice"]["gross"]["amount"] == float(
-        product_variant_price
+    order_line1 = order_data["lines"][0]
+    order_line2 = order_data["lines"][1]
+    order_line3 = order_data["lines"][2]
+
+    assert len(order_data["lines"]) == 3
+
+    assert order_line1["quantity"] == product2_new_quantity
+    assert order_line1["unitPrice"]["gross"]["amount"] == line1_unit_price
+    assert order_line1["unitDiscount"]["amount"] == line1_discount
+    assert order_line1["unitDiscountReason"] == f"Sale: {sale_id}"
+    assert (
+        order_line1["undiscountedUnitPrice"]["gross"]["amount"]
+        == product2_variant_price
+    )
+
+    assert order_line2["quantity"] == product3_new_quantity
+    assert order_line2["unitPrice"]["gross"]["amount"] == line2_unit_price
+    assert order_line2["unitDiscount"]["amount"] == line2_discount
+    assert order_line2["unitDiscountReason"] == f"Sale: {sale_id}"
+    assert (
+        order_line2["undiscountedUnitPrice"]["gross"]["amount"]
+        == product3_variant_price
     )
-    assert order_line["unitDiscountType"] == "FIXED"
-    assert order_line["unitPrice"]["gross"]["amount"] == unit_price
-    assert order_line["unitDiscount"]["amount"] == line_discount
-    assert order_line["unitDiscountReason"] == f"Sale: {sale_id}"
+
+    assert order_line3["quantity"] == product4_quantity
+    assert order_line3["unitPrice"]["gross"]["amount"] == line3_unit_price
+    assert order_line3["unitDiscount"]["amount"] == 0
+    assert order_line3["unitDiscountReason"] is None
+    assert (
+        order_line3["undiscountedUnitPrice"]["gross"]["amount"]
+        == product4_variant_price
+    )
+
+    assert order_data["total"]["gross"]["amount"] == calculated_total
+    assert order_data["subtotal"]["gross"]["amount"] == calculated_subtotal
```

Comment:
Maybe we should also check if the sum of the returned line total prices is equal to `order_data["subtotal"]["gross"]["amount"]`